src/code/audio_synthesis.c

```diff
@@ -832,13 +831,14 @@ Acmd* AudioSynth_ProcessNote(s32 noteIndex, NoteSubEu* noteSubEu, NoteSynthesisS
                 nSamplesToProcess = samplesLenAdjusted - nSamplesProcessed;
 
                 if (nFirstFrameSamplesToIgnore == 0 && !synthState->restart) {
-                    nFirstFrameSamplesToIgnore = 16;
+                    nFirstFrameSamplesToIgnore = SAMPLES_PER_FRAME;
                 }
-                nSamplesInFirstFrame = 16 - nFirstFrameSamplesToIgnore;
+                nSamplesInFirstFrame = SAMPLES_PER_FRAME - nFirstFrameSamplesToIgnore;
 
                 if (nSamplesToProcess < nSamplesUntilLoopEnd) {
-                    nFramesToDecode = (s32)(nSamplesToProcess - nSamplesInFirstFrame + 15) / 16;
-                    nSamplesToDecode = nFramesToDecode * 16;
+                    nFramesToDecode =
+                        (s32)(nSamplesToProcess - nSamplesInFirstFrame + (SAMPLES_PER_FRAME - 1)) / SAMPLES_PER_FRAME;
```

Comment:
```suggestion
                        (s32)(nSamplesToProcess - nSamplesInFirstFrame + SAMPLES_PER_FRAME - 1) / SAMPLES_PER_FRAME;
```
should match too
same elsewhere